Validation/EcalDigis/plugins/EcalSelectiveReadoutValidation.h

```diff
@@ -789,13 +789,15 @@ class EcalSelectiveReadoutValidation : public DQMOneEDAnalyzer<> {
   /** Energy deposited in ECAL barrel crystals. Eta index starts from 0 at
    * eta minimum and phi index starts at phi=0+ in CMS std coordinate system.
    */
-  energiesEb_t ebEnergies[nEbEta][nEbPhi];
+  std::vector<std::vector<energiesEb_t>> ebEnergies =
+      std::vector<std::vector<energiesEb_t>>(nEbEta, std::vector<energiesEb_t>(nEbPhi));
 
   /** Energy deposited in ECAL endcap crystals. Endcap index is 0 for EE- and
    * 1 for EE+. X and Y index starts at x and y minimum in std CMS coordinate
    * system.
    */
-  energiesEe_t eeEnergies[nEndcaps][nEeX][nEeY];
+  std::vector<std::vector<std::vector<energiesEe_t>>> eeEnergies = std::vector<std::vector<std::vector<energiesEe_t>>>(
+      nEndcaps, std::vector<std::vector<energiesEe_t>>(nEeX, std::vector<energiesEe_t>(nEeY)));
 
```

Comment:
Why is std::vector better than C array? Why not std::array?

Reply:
Please see reply above

Reply:
It looks like that in both cases an std::array of std::arrays ought to work and save all the memory overhead of vectors of vectors.

Reply:
I tried with std::array of std::arrays but the run time error re-appears.